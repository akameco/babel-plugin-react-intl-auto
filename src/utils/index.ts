import { sep } from 'path'
import * as t from '@babel/types'
import murmur from 'murmurhash3js'
import { NodePath } from '@babel/core'

const REG = new RegExp(`\\${sep}`, 'gu')

const isObjectProperties = (
  properties: NodePath[]
): properties is NodePath<t.ObjectProperty>[] =>
  properties.every(p => p.isObjectProperty())

export const createHash = (message: string) => `${murmur.x86.hash32(message)}`

export const dotPath = (str: string, separator: string | undefined = '.') =>
  str.replace(REG, separator)

export const objectProperty = (
  key: string,
  value: string | t.StringLiteral | t.TemplateLiteral
) => {
  const valueNode = typeof value === 'string' ? t.stringLiteral(value) : value
  return t.objectProperty(t.stringLiteral(key), valueNode)
}

export function getObjectProperties(path: NodePath) {
  if (path.isObjectExpression()) {
    const properties = path.get('properties')
    if (isObjectProperties(properties)) {
      return properties
    }
  } else if (path.isIdentifier()) {
    const binding = path.scope.getBinding(path.node.name)
    if (!binding) {
      return null
    }
    const properties = binding.path.get('init.properties')
    if (Array.isArray(properties) && isObjectProperties(properties)) {
      return properties
    }
  }
  return null
}
