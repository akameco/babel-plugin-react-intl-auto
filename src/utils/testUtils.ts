// eslint-disable-next-line import/no-extraneous-dependencies
import pluginTester from 'babel-plugin-tester'
import { Opts } from '../types'
import plugin from '..'

export function cases(
  filename: string,
  testCases: {
    title: string
    tests: { title: string; code: string }[]
    pluginOptions?: Opts
  }[]
) {
  const defaultOpts = {
    title: '',
    plugin,
    snapshot: true,
    babelOptions: { filename, parserOpts: { plugins: ['jsx'] } },
    formatResult: (r: any) => r,
    tests: [],
  }

  for (const testCase of testCases) {
    testCase.tests = testCase.tests.map(t => ({ ...t, title: t.title }))
    pluginTester({ ...defaultOpts, ...testCase })
  }
}
