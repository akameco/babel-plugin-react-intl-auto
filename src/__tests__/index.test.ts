import path from 'path'
import { cases } from '../utils/testUtils'

const filename = path.resolve(__dirname, '..', '__fixtures__', 'messages.js')

const defaultTest = {
  title: 'default',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})
`,
}

const objectTest = {
  title: 'Object',
  code: `
import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})
`,
}

const multiExportTest = {
  title: 'multi export',
  code: `
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})
`,
}

const leadingCommentTest = {
  title: 'leading comment',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages({
  // The main Hello of our app.
  hello: 'hello',

  // Another Hello,
  // multiline this time
  world: {
    id: 'hello.world',
    defaultMessage: 'hello world',
  }
})
`,
}

const leadingCommentWithDescriptionTest = {
  title: 'leading comment with description',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages({

  // This comment should not be used
  world: {
    defaultMessage: 'hello world',
    description: 'The hello world',
  }
})
`,
}

const evalStringTest = {
  title: 'eval string',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello' + 'world',
})
`,
}

const tests = [
  defaultTest,
  {
    title: 'with include value',
    code: `
import { defineMessages } from 'react-intl'

defineMessages({
  hello: \`hello world \${1}\`,
})
      `,
  },

  {
    title: 'string literal',
    code: `
import { defineMessages } from 'react-intl'

defineMessages({
  'hello': 'hello world'
})
      `,
  },

  objectTest,
  {
    title: 'import as',
    code: `
import { defineMessages as m } from 'react-intl'

m({
  hello: 'hello'
})

`,
  },

  {
    title: 'with other func',
    code: `
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
})

hello({
  id: 'hoge',
})
    `,
  },

  multiExportTest,
  {
    title: 'throw error when key is NumiricLiteral',
    code: `
import { defineMessages } from 'react-intl'

export default defineMessages({
  1: 'hello',
})
`,
    error: /requires Object key or string literal/u,
    snapshot: false,
  },

  {
    title: 'not transform if defineMessages is not imported',
    code: `
import any from 'any-module'

export default defineMessages({
  hello: 'hello'
})
    `,
  },

  {
    title: 'not transform when defineMessages argumens is not object',
    code: `
import { defineMessages } from 'react-intl'

export default defineMessages(1)
    `,
  },

  {
    title: 'when using the variable',
    code: `
import { defineMessages } from 'react-intl'

const messages = {hello: 'hello'}

export default defineMessages(messages)
    `,
  },

  {
    title: 'not transfrom when the variable can not be found',
    code: `
import { defineMessages } from 'react-intl'

export default defineMessages(messages)
    `,
  },

  {
    title: 'not transform when defineMessages argumens is empty',
    code: `
import { defineMessages } from 'react-intl'

export default defineMessages()
    `,
  },

  {
    title: 'not transform if callee is not identifier',
    code: `
import { defineMessages } from 'react-intl'

const m = [defineMessages]

export default m[0]({
  hello: 'hello world'
})
    `,
  },

  {
    title: 'with other specifier',
    code: `
import { defineMessages, FormattedMessage } from 'react-intl'

export default defineMessages({
  hello: 'hello world',
})
    `,
  },

  leadingCommentTest,
  leadingCommentWithDescriptionTest,
  evalStringTest,
]

const moduleSourceNameTest = {
  title: 'moduleSourceName',
  code: `
import { defineMessages } from 'gatsby-plugin-intl'

export default defineMessages({
  hello: 'hello',
})
  `,
}

cases(filename, [
  { title: 'default', tests },
  {
    title: 'removePrefix = "src"',
    tests: [defaultTest],
    pluginOptions: { removePrefix: 'src' },
  },

  {
    title: 'removePrefix = "src/" -- with slash',
    tests: [defaultTest],
    pluginOptions: { removePrefix: 'src/' },
  },

  {
    title: 'filebase = true',
    tests: [defaultTest],
    pluginOptions: { filebase: true },
  },

  {
    title: 'includeExportName = true',
    tests: [defaultTest, multiExportTest],
    pluginOptions: { includeExportName: true },
  },

  {
    title: 'includeExportName = all',
    tests: [defaultTest, multiExportTest],
    pluginOptions: { includeExportName: 'all' },
  },

  {
    title: 'removePrefix = true, includeExportName = true',
    tests: [defaultTest, multiExportTest],
    pluginOptions: { removePrefix: true, includeExportName: true },
  },

  {
    title: 'removePrefix = false',
    tests: [defaultTest, multiExportTest],
    pluginOptions: { removePrefix: false },
  },

  {
    title: 'removePrefix = true, includeExportName = all',
    tests: [defaultTest, multiExportTest],
    pluginOptions: { removePrefix: true, includeExportName: 'all' },
  },

  {
    title: 'extractComments = false',
    tests: [defaultTest, leadingCommentTest, leadingCommentWithDescriptionTest],
    pluginOptions: { extractComments: false },
  },

  {
    title: 'removePrefix = /__fixtures__/',
    tests: [defaultTest],
    pluginOptions: {
      removePrefix: /src[\\/]__f.+?_/u,
      includeExportName: true,
    },
  },

  {
    title: 'removePrefix = "src.__fixtures__"',
    tests: [defaultTest],
    pluginOptions: {
      removePrefix: 'src.__fixtures__',
    },
  },

  {
    title: 'removePrefix = "src.__fixtures__", includeExportName = true',
    tests: [defaultTest, multiExportTest],
    pluginOptions: {
      removePrefix: 'src.__fixtures__',
      includeExportName: true,
    },
  },

  {
    title: 'moduleSourceNameTest',
    tests: [defaultTest, moduleSourceNameTest],
    pluginOptions: {
      moduleSourceName: 'gatsby-plugin-intl',
    },
  },

  {
    title: 'separator = ""',
    tests: [defaultTest, multiExportTest],
    pluginOptions: {
      separator: '',
    },
  },

  {
    title: 'separator = "_"',
    // tests: [defaultTest, multiExportTest],
    tests: [defaultTest, multiExportTest],
    pluginOptions: {
      separator: '_',
    },
  },

  {
    title: 'separator = "foo"',
    tests: [defaultTest, multiExportTest],
    pluginOptions: {
      separator: 'foo',
    },
  },

  {
    title: 'relativeTo = "../"',
    tests: [defaultTest, multiExportTest],
    pluginOptions: {
      relativeTo: '..',
    },
  },

  {
    title: 'relativeTo = ""',
    tests: [defaultTest, multiExportTest],
    pluginOptions: {
      relativeTo: '',
    },
  },

  {
    title: 'removeDefaultMessage = true',
    tests: [defaultTest, objectTest, evalStringTest],
    pluginOptions: {
      removeDefaultMessage: true,
    },
  },
])
