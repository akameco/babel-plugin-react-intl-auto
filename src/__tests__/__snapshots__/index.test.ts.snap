// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default Object: Object 1`] = `

import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    "id": "src.__fixtures__.world",
    defaultMessage: 'world'
  },
  headerTitle: {
    "id": "src.__fixtures__.headerTitle",
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});

`;

exports[`default default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`default eval string: eval string 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello' + 'world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.helloworld",
    "defaultMessage": "helloworld"
  }
});

`;

exports[`default import as: import as 1`] = `

import { defineMessages as m } from 'react-intl'

m({
  hello: 'hello'
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages as m } from 'react-intl';
m({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`default leading comment with description: leading comment with description 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({

  // This comment should not be used
  world: {
    defaultMessage: 'hello world',
    description: 'The hello world',
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  // This comment should not be used
  world: {
    "id": "src.__fixtures__.world",
    defaultMessage: 'hello world',
    description: 'The hello world'
  }
});

`;

exports[`default leading comment: leading comment 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  // The main Hello of our app.
  hello: 'hello',

  // Another Hello,
  // multiline this time
  world: {
    id: 'hello.world',
    defaultMessage: 'hello world',
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  // The main Hello of our app.
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello',
    "description": "The main Hello of our app."
  },
  // Another Hello,
  // multiline this time
  world: {
    id: 'hello.world',
    defaultMessage: 'hello world',
    "description": "Another Hello,\\nmultiline this time"
  }
});

`;

exports[`default multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`default not transform if callee is not identifier: not transform if callee is not identifier 1`] = `

import { defineMessages } from 'react-intl'

const m = [defineMessages]

export default m[0]({
  hello: 'hello world'
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
const m = [defineMessages];
export default m[0]({
  hello: 'hello world'
});

`;

exports[`default not transform if defineMessages is not imported: not transform if defineMessages is not imported 1`] = `

import any from 'any-module'

export default defineMessages({
  hello: 'hello'
})

      ↓ ↓ ↓ ↓ ↓ ↓

import any from 'any-module';
export default defineMessages({
  hello: 'hello'
});

`;

exports[`default not transform when defineMessages argumens is empty: not transform when defineMessages argumens is empty 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages()

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages();

`;

exports[`default not transform when defineMessages argumens is not object: not transform when defineMessages argumens is not object 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages(1)

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages(1);

`;

exports[`default not transfrom when the variable can not be found: not transfrom when the variable can not be found 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages(messages)

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages(messages);

`;

exports[`default string literal: string literal 1`] = `

import { defineMessages } from 'react-intl'

defineMessages({
  'hello': 'hello world'
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
defineMessages({
  'hello': {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`default when using the variable: when using the variable 1`] = `

import { defineMessages } from 'react-intl'

const messages = {hello: 'hello'}

export default defineMessages(messages)

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
const messages = {
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
};
export default defineMessages(messages);

`;

exports[`default with include value: with include value 1`] = `

import { defineMessages } from 'react-intl'

defineMessages({
  hello: \`hello world \${1}\`,
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": \`hello world \${1}\`
  }
});

`;

exports[`default with other func: with other func 1`] = `

import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
})

hello({
  id: 'hoge',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});
hello({
  id: 'hoge'
});

`;

exports[`default with other specifier: with other specifier 1`] = `

import { defineMessages, FormattedMessage } from 'react-intl'

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages, FormattedMessage } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`extractComments = false default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`extractComments = false leading comment with description: leading comment with description 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({

  // This comment should not be used
  world: {
    defaultMessage: 'hello world',
    description: 'The hello world',
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  // This comment should not be used
  world: {
    "id": "src.__fixtures__.world",
    defaultMessage: 'hello world',
    description: 'The hello world'
  }
});

`;

exports[`extractComments = false leading comment: leading comment 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  // The main Hello of our app.
  hello: 'hello',

  // Another Hello,
  // multiline this time
  world: {
    id: 'hello.world',
    defaultMessage: 'hello world',
  }
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  // The main Hello of our app.
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  },
  // Another Hello,
  // multiline this time
  world: {
    id: 'hello.world',
    defaultMessage: 'hello world'
  }
});

`;

exports[`filebase = true default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.messages.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`includeExportName = all default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.default.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`includeExportName = all multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "src.__fixtures__.extra.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.default.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`includeExportName = true default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`includeExportName = true multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "src.__fixtures__.extra.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`moduleSourceNameTest default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: 'hello'
});

`;

exports[`moduleSourceNameTest moduleSourceName: moduleSourceName 1`] = `

import { defineMessages } from 'gatsby-plugin-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'gatsby-plugin-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`relativeTo = "" default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`relativeTo = "" multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`relativeTo = "../" default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "babel-plugin-react-intl-auto.src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`relativeTo = "../" multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "babel-plugin-react-intl-auto.src.__fixtures__.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "babel-plugin-react-intl-auto.src.__fixtures__.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`removeDefaultMessage = true Object: Object 1`] = `

import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
defineMessages({
  new: {
    id: 'this is id'
  },
  world: {
    "id": "src.__fixtures__.world"
  },
  headerTitle: {
    "id": "src.__fixtures__.headerTitle",
    description: 'Message to greet the user.'
  }
});

`;

exports[`removeDefaultMessage = true default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello"
  }
});

`;

exports[`removeDefaultMessage = true eval string: eval string 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello' + 'world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.helloworld"
  }
});

`;

exports[`removePrefix = "src" default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = "src.__fixtures__" default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = "src.__fixtures__", includeExportName = true default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = "src.__fixtures__", includeExportName = true multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "extra.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`removePrefix = "src/" -- with slash default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = /__fixtures__/ default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "_.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = false default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = false multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "src.__fixtures__.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`removePrefix = true, includeExportName = all default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "default.hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = true, includeExportName = all multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "extra.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "default.hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`removePrefix = true, includeExportName = true default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`removePrefix = true, includeExportName = true multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "extra.hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`separator = "" default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src__fixtures__hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`separator = "" multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "src__fixtures__hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "src__fixtures__hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`separator = "_" default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "src___fixtures___hello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`separator = "_" multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "src___fixtures___hello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "src___fixtures___hello",
    "defaultMessage": 'hello world'
  }
});

`;

exports[`separator = "foo" default: default 1`] = `

import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export default defineMessages({
  hello: {
    "id": "srcfoo__fixtures__foohello",
    "defaultMessage": 'hello'
  }
});

`;

exports[`separator = "foo" multi export: multi export 1`] = `

import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello world extra'
})

export default defineMessages({
  hello: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';
export const extra = defineMessages({
  hello: {
    "id": "srcfoo__fixtures__foohello",
    "defaultMessage": 'hello world extra'
  }
});
export default defineMessages({
  hello: {
    "id": "srcfoo__fixtures__foohello",
    "defaultMessage": 'hello world'
  }
});

`;
